Added documentation for label painting

BhavyaC16 opened the pull request:
# Description
Added clarification that `add_labels` modifies the input array.
Provided notes at the end of docstring, suggesting that a copy can be made to avoid modification of the input array.
Fixes #1041 

## Type of change
- [x] Documentation

BhavyaC16 committed:
Added documentation for label painting

Added clarification that add_labels modifies input array.

sofroniewn reviewed (approved):
This is great @BhavyaC16, thanks for your first contribution! It looks ready to go for me, but I'll let @jni take a quick look before merge too

jni commented on napari/components/add_layers_mixin.py:
```diff
@@ -394,6 +394,8 @@ def add_labels(
         An image-like layer where every pixel contains an integer ID
         corresponding to the region it belongs to.
 
+        Modifies the input array in-place.
```
This isn't necessarily true, if one is only looking at the layer. I would say "Using the viewer's label editing tools (painting, erasing) will modify the input-array in-place. To avoid this, pass a copy as follows:"

You might also specify that you can *get* the painted labels back with:

```python
layer = viewer.add_labels(data.copy())
# do some painting/editing
result = layer.data`. 
```

jni reviewed (changes requested):
@BhavyaC16 thank you for taking this on! I've made a suggestion to improve the text a little. I also think that the "modifies" and "how to avoid it" notes should be together. I think I prefer them at the top of the docstring rather than in a Notes section, because it is very important that users see it.

Additionally, could you please make the corresponding changes to `napari.view_labels` as well?

BhavyaC16 committed:
Added documentation for view_labels

Added clarification for modification of input-array in-place when using label editing tools (painting/erasing).

BhavyaC16 committed:
Updated documentation for add_labels

Updated clarification for modification of input-array in-place when using label editing tools (painting/erasing).
